CoseSignTool/Local/HeaderStringConverter.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace CoseSignTool.Local;
+
+internal class HeaderStringConverter : JsonConverter
+{
+    public override bool CanConvert(Type objectType)
+    {
+        return objectType == typeof(string);
+    }
+
+    /// <summary>
+    /// Ensure that the value is not null or empty.
+    /// </summary>
+    /// <param name="reader"></param>
+    /// <param name="objectType"></param>
+    /// <param name="existingValue"></param>
+    /// <param name="serializer"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    public override object? ReadJson(JsonReader reader, Type objectType, object? existingValue, JsonSerializer serializer)
+    {
+        return reader.Value == null || string.IsNullOrEmpty(reader.Value.ToString()) ? throw new ArgumentNullException("String header value cannot be null or empty.") : reader.Value.ToString();
+    }
+
+    public override void WriteJson(JsonWriter writer, object? value, JsonSerializer serializer)
+    {
+        throw new NotImplementedException();
```

Comment:
You have plans to implement this later?

Reply:
No. I only plan to do validation during deserialization.